compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -136,9 +137,14 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         }
 
         let mut reported_spans = FxHashSet::default();
-        for error in &self.privacy_errors {
-            if reported_spans.insert(error.dedup_span) {
-                self.report_privacy_error(error);
+        self.privacy_errors.sort_by_key(|e| e.ident.span);
+        for error in std::mem::take(&mut self.privacy_errors) {
```

Comment:
Because `fn report_privacy_error(&mut`, which is needed to be able to call `lookup_import_candidates` from within it, iterating over `privacy_errors` without taking ownership of them is not possible. I thought that `privacy_errors` was not being used for anything after the errors were reported. Was this understanding wrong?

Reply:
Ah, okay, the signature of `report_privacy_error` was changed to `&mut self`, I didn't notice.
I though the change ended up being unnecessary.
(`privacy_errors` can be taken out at this point, yes.)